src/execution/experimental-executions.js

```diff
@@ -523,66 +679,62 @@ function getWhere(filters) {
 
 const sortToOrderBy = item => ({ column: get(item, 'element'), direction: get(item, 'sort') });
 
-const getOrderBy = (metrics, categories, type) => {
-    // For bar chart we always override sorting to sort by values (first metric)
-    if (type === 'bar' && notEmpty(metrics)) {
-        return [{
-            column: first(compact(map(metrics, 'element'))),
-            direction: 'desc'
-        }];
-    }
-
+const getOrderBy = (metrics, categories) => {
     return map(filter([...categories, ...metrics], item => item.sort), sortToOrderBy);
 };
 
-export const mdToExecutionConfiguration = (mdObj, options = {}) => {
-    const buckets = get(mdObj, 'buckets');
-    const measures = map(buckets.measures, ({ measure }) => measure);
-    const metrics = flatten(map(measures, (measure, index) => getMetricFactory(measure)(measure, buckets, index)));
-
-    let categories = getCategories(buckets);
-    let filters = getFilters(buckets);
-    if (options.removeDateItems) {
-        categories = filter(categories, ({ category }) => category.type !== 'date');
-        filters = filter(filters, item => !item.dateFilter);
+export function loadAttributesMap(projectId, categoryDisplayForms) {
+    if (categoryDisplayForms.length === 0) {
+        return Promise.resolve({});
     }
-    categories = map(categories, categoryToElement);
 
-    const columns = compact(map([...categories, ...metrics], 'element'));
-
-    return {
-        columns,
-        orderBy: getOrderBy(metrics, categories, get(mdObj, 'type')),
-        definitions: sortDefinitions(compact(map(metrics, 'definition'))),
-        where: columns.length ? getWhere(filters) : {},
-        metricMappings: map(metrics, m => ({ element: m.element, ...m.meta }))
-    };
-};
-
-const getOriginalMetricFormats = (mdObj) => {
-    // for metrics with showPoP or measureFilters.length > 0 roundtrip for original metric format
-    return Promise.all(map(
-        map(get(mdObj, 'buckets.measures'), ({ measure }) => measure),
-        (measure) => {
-            if (measure.showPoP === true || measure.measureFilters.length > 0) {
-                return xhrGet(measure.objectUri).then((obj) => {
-                    return {
-                        ...measure,
-                        format: get(obj, 'metric.content.format', measure.format)
-                    };
-                });
-            }
+    return getObjects(projectId, categoryDisplayForms).then((displayForms) => {
```

Comment:
[Q] Are you sure by the naming of the `categoryDisplayForms` parameter?

* What does the `category` say us?
* It suggest that display form object is being passed into the function, while I believe it's just display form *uris*

Reply:
renamed